common/djangoapps/student/tasks.py

```diff
@@ -108,9 +114,11 @@ def send_course_enrollment_email(
             }
         )
     except Exception:  # pylint: disable=broad-except
+        segment_properties.update({'is_course_run_available': False})
         log.exception(
             f"Unable to get data for user {user_id} from discovery for course {course_id}"
         )
+    segment.track(user_id, 'edx.course.enrollment.email.data.sections', segment_properties)
```

Comment:
We shouldn't send it for all users. We should only send it if either or both the course run or course data is missing